### PR TITLE
fix(async-csv): Finishing touches on some async-csv bugs

### PR DESCRIPTION
**Pull request by leeandher**
This PR is aimed at fixing up 4 smaller tasks.

1. When the user refreshes the page and clicks 'Export' again, we return a 200 (not a 201) because we didn't queue up another export. We just found their previous. Instead of swallowing this, the user now receives a different toast notification

2. As this will feature will likely go to GA soon, we're going to temporarily enforce a hard limit on file sizes at 1M rows.

3. The 404 response from the server doesn't have a `responseJSON` field. This change should still render an alright UI in case this happens (resolves JAVASCRIPT-222B)

4. Increase the number of entries written to the CSV every iteration from 1000 to something higher. I just set it to <s>50000</s> 10000 as a safety net.

### CODE REVIEW

File: src/sentry/data_export/tasks.py [hunk 1]
```diff
@@ -22,7 +22,7 @@
 
 
 @instrumented_task(name="sentry.data_export.tasks.assemble_download", queue="data_export")
-def assemble_download(data_export_id, limit=None, environment_id=None):
+def assemble_download(data_export_id, limit=1000000, environment_id=None):
```
Comment (leeandher) [inline src/sentry/data_export/tasks.py:25]: This is point 2. Enforces a hard limit of 1M rows

File: src/sentry/static/sentry/app/components/dataExport.tsx [hunk 2]
```diff
@@ -46,18 +46,22 @@ class DataExport extends React.Component<Props, State> {
...
             query_info: queryInfo,
           },
         }
       );
+      const {id: dataExportId} = data;
       addSuccessMessage(
-        t("Sit tight. We'll shoot you an email when your data is ready for download.")
+        response?.status === 201
+          ? t("Sit tight. We'll shoot you an email when your data is ready for download.")
+          : t("It looks like we're already working on it. Sit tight, we'll email you.")
```
Comment (leeandher) [inline src/sentry/static/sentry/app/components/dataExport.tsx:64]: This is Point 1. Show a different toast message when the user refreshes the page and tries the same download. (There is a ticket to eventually save the state across refreshes, but it proves a little more difficult, so this is a temp fix)

File: src/sentry/static/sentry/app/views/dataExport/dataDownload.tsx [hunk 3]
```diff
@@ -174,9 +175,11 @@ class DataDownload extends AsyncView<Props, State> {
...
             </h3>
           </Header>
-          <Body>
-            <p>{err.responseJSON.detail}</p>
-          </Body>
+          {errDetail && (
+            <Body>
+              <p>{errDetail}</p>
+            </Body>
+          )}
```
Comment (leeandher) [inline src/sentry/static/sentry/app/views/dataExport/dataDownload.tsx:182]: This is Point 3. If there isn't an error message, don't show this body section (as is the case in 404s)

File: src/sentry/data_export/base.py [hunk 4]
```diff
@@ -4,7 +4,7 @@
 from datetime import timedelta
 from enum import Enum
 
-SNUBA_MAX_RESULTS = 1000
+SNUBA_MAX_RESULTS = 50000
```
Comment (leeandher) [inline src/sentry/data_export/base.py:7]: Allow for more results per iteration (speeds up task by 50x). I believe the only limitation maybe memory issues by have a list that long with `dict`s in every spot. Not entirely sure on the memory limitations of workers in production so I think this is a good spot for now.

Comment (leeandher) [reply src/sentry/data_export/base.py]: Changed to 10k, since that's the actual max result limit. My bad!

**Review by MeredithAnya — APPROVED** [review]

**Pull request merged**